rest_framework/relations.py

```diff
@@ -492,10 +492,19 @@ def to_internal_value(self, data):
         if not self.allow_empty and len(data) == 0:
             self.fail('empty')
 
-        return [
-            self.child_relation.to_internal_value(item)
-            for item in data
-        ]
+        if isinstance(self.child_relation, PrimaryKeyRelatedField):
+            values = list(self.child_relation.get_queryset().filter(pk__in=data))
+            missing_primary_keys = set(v.pk for v in values) - set(data)
```

Comment:
We are not bothered about distinct values, `Model.objects.filter(pk__in=[1, 1])` will only return 1 model AFAIK. We use `set()` here to just easily find any differences between the returned list of primary keys and the given list of primary keys.